subprojects/core/src/main/java/org/gradle/internal/installation/GradleRuntimeShadedJarDetector.java

```diff
@@ -62,28 +62,11 @@ private static boolean isJarUrl(URL location) {
     }
 
     private static boolean findMarkerFileInJar(File jar) {
-        JarFile jarFile = null;
-
-        try {
-            jarFile = new JarFile(jar);
+        try (JarFile jarFile = new JarFile(jar)) {
             JarEntry markerFile = jarFile.getJarEntry(MARKER_FILENAME);
-
-            if (markerFile != null) {
-                return true;
-            }
+            return markerFile != null;
         } catch (IOException e) {
             throw UncheckedException.throwAsUncheckedException(e);
-        } finally {
-            if (jarFile != null) {
-                try {
-                    jarFile.close();
-                } catch (IOException e) {
-                    //noinspection ThrowFromFinallyBlock
```

Comment:
this the `Finally` rule, kind of extending the scope...

Should we extract this, or include to avoid whole pipeline?